leases/schema/mutations.py

```diff
@@ -167,6 +167,149 @@ def mutate_and_get_payload(cls, root, info, **input):
         return DeleteBerthLeaseMutation()
 
 
+class CreateWinterStorageLeaseMutation(graphene.ClientIDMutation):
+    class Input(AbstractLeaseInput):
+        application_id = graphene.ID(required=True)
+        place_id = graphene.ID(required=True)
+
+    winter_storage_lease = graphene.Field(WinterStorageLeaseNode)
+
+    @classmethod
+    @view_permission_required(WinterStorageApplication, CustomerProfile)
+    @add_permission_required(WinterStorageLease)
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **input):
+        application = get_node_from_global_id(
+            info,
+            input.pop("application_id"),
+            only_type=WinterStorageApplicationNode,
+            nullable=False,
+        )
+
+        if not application.customer:
+            raise VenepaikkaGraphQLError(
+                _("Application must be connected to an existing customer first")
+            )
+
+        place = get_node_from_global_id(
+            info,
+            input.pop("place_id"),
+            only_type=WinterStoragePlaceNode,
+            nullable=False,
+        )
+
+        input["application"] = application
+        input["place"] = place
+        input["customer"] = application.customer
+
+        if input.get("boat_id", None):
+            from customers.schema import BoatNode
+
+            boat = get_node_from_global_id(
+                info, input.pop("boat_id"), only_type=BoatNode, nullable=False,
+            )
+
+            if boat.owner.id != input["customer"].id:
+                raise VenepaikkaGraphQLError(
+                    _("Boat does not belong to the same customer as the Application")
+                )
+
+            input["boat"] = boat
+
+        try:
+            lease = WinterStorageLease.objects.create(**input)
+        except ValidationError as e:
+            # Flatten all the error messages on a single list
+            errors = sum(e.message_dict.values(), [])
+            raise VenepaikkaGraphQLError(errors)
+
+        application.status = ApplicationStatus.OFFER_GENERATED
```

Comment:
we use `PENDING` status for the berth leases 🤔 `OFFER_GENERATED` means there is an `Order` already assigned to the lease, but currently lease creation does not generate order. there is a ticket that it should ([VEN-695](https://helsinkisolutionoffice.atlassian.net/browse/VEN-695)). maybe then we can revisit the proper statuses for this in each of the mutation, but for now `PENDING`? 🤔

Reply:
alright, my bad, scratch this. this is the correct `ApplicationStatus`, I was thinking about `LeaseStatus`